### PR TITLE
Update for OS 14 Public Beta

### PR DESCRIPTION
**Pull request by Steve-Glass**
# Description
New tool, Bug fixing, or Improvement?
Please include a summary of the change and which issue is fixed. Also include relevant motivation and context.
**For new tools, please provide total size and installation time.**

## Check list
- [ ] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

**Review by mikhailkoliada — DISMISSED** [review]

**Review by erik-bershel — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -19,15 +19,16 @@ To build a VM machine from this repo's source, see the [instructions](docs/creat
...
-| Windows Server 2022 | `windows-latest` or `windows-2022` | [windows-2022] | [![statuswin22](https://gh-runnerimagesdeploymentstatus.azurewebsites.net/api/status?imageName=windows-2022&badge=1)](https://gh-runnerimagesdeploymentstatus.azurewebsites.net/api/status?imageName=windows-2022&redirect=1) |
-| Windows Server 2019 | `windows-2019` | [windows-2019] | [![statuswin19](https://gh-runnerimagesdeploymentstatus.azurewebsites.net/api/status?imageName=windows-2019&badge=1)](https://gh-runnerimagesdeploymentstatus.azurewebsites.net/api/status?imageName=windows-2019&redirect=1)
+| Image | YAML Label | Included Software | 
+| --------------------|---------------------|--------------------|
+| Ubuntu 22.04 | `ubuntu-latest` or `ubuntu-22.04` | [ubuntu-22.04] 
+| Ubuntu 20.04 | `ubuntu-20.04` | [ubuntu-20.04] | 
+| macOS 14 [beta] | `macos-14`, `macos-14-large`, or  `macos-14-xlarge`| [macOS-14] |
+| macOS 13 | `macos-13`, `macos-13-large`, or  `macos-13-xlarge`| [macOS-13] |
+| macOS 12 | `macos-latest`,`macos-12`, `macos-latest-large`, `macos-12-large`, or `macos-12-large`| [macOS-12] |
```
Comment (Vyazovoy) [inline README.md:28]: @Steve-Glass, you have `macos-12-large` twice here. Did you mean `xlarge` as the last item?

Comment (Steve-Glass) [reply README.md]: Good catch and updated. I removed the duplicate as there is no `macos-12-xlarge` runner.

File: README.md [hunk 2]
```diff
@@ -19,15 +19,16 @@ To build a VM machine from this repo's source, see the [instructions](docs/creat
...
-| macOS 11 | `macos-11`| [macOS-11] | [![statusmac11](https://gh-runnerimagesdeploymentstatus.azurewebsites.net/api/status?imageName=macos-11&badge=1)](https://gh-runnerimagesdeploymentstatus.azurewebsites.net/api/status?imageName=macos-11&redirect=1)
-| Windows Server 2022 | `windows-latest` or `windows-2022` | [windows-2022] | [![statuswin22](https://gh-runnerimagesdeploymentstatus.azurewebsites.net/api/status?imageName=windows-2022&badge=1)](https://gh-runnerimagesdeploymentstatus.azurewebsites.net/api/status?imageName=windows-2022&redirect=1) |
-| Windows Server 2019 | `windows-2019` | [windows-2019] | [![statuswin19](https://gh-runnerimagesdeploymentstatus.azurewebsites.net/api/status?imageName=windows-2019&badge=1)](https://gh-runnerimagesdeploymentstatus.azurewebsites.net/api/status?imageName=windows-2019&redirect=1)
+| Image | YAML Label | Included Software | 
+| --------------------|---------------------|--------------------|
+| Ubuntu 22.04 | `ubuntu-latest` or `ubuntu-22.04` | [ubuntu-22.04]
```
Comment (muzimuzhi) [inline README.md:24]: Missing trailing vert (`|`)
```suggestion
| Ubuntu 22.04 | `ubuntu-latest` or `ubuntu-22.04` | [ubuntu-22.04] |
```

File: README.md [hunk 3]
```diff
@@ -19,15 +19,16 @@ To build a VM machine from this repo's source, see the [instructions](docs/creat
...
+| Image | YAML Label | Included Software | 
+| --------------------|---------------------|--------------------|
+| Ubuntu 22.04 | `ubuntu-latest` or `ubuntu-22.04` | [ubuntu-22.04] 
+| Ubuntu 20.04 | `ubuntu-20.04` | [ubuntu-20.04] | 
+| macOS 14 [beta] | `macos-14`, `macos-14-large`, or  `macos-14-xlarge`| [macOS-14] |
+| macOS 13 | `macos-13`, `macos-13-large`, or  `macos-13-xlarge`| [macOS-13] |
+| macOS 12 | `macos-latest`,`macos-12`, `macos-latest-large`, or `macos-12-large`| [macOS-12] | 
+| macOS 11 | `macos-11`| [macOS-11] | 
+| Windows Server 2022 | `windows-latest` or `windows-2022` | [windows-2022] | 
+| Windows Server 2019 | `windows-2019` | [windows-2019] | )
```
Comment (muzimuzhi) [inline README.md:31]:
```suggestion
| Windows Server 2019 | `windows-2019` | [windows-2019] |
```

**Review by muzimuzhi — CHANGES_REQUESTED** [review]
Caught some table formatting typos.

File: README.md [hunk 4]
```diff
@@ -19,15 +19,16 @@ To build a VM machine from this repo's source, see the [instructions](docs/creat
...
-| Windows Server 2019 | `windows-2019` | [windows-2019] | [![statuswin19](https://gh-runnerimagesdeploymentstatus.azurewebsites.net/api/status?imageName=windows-2019&badge=1)](https://gh-runnerimagesdeploymentstatus.azurewebsites.net/api/status?imageName=windows-2019&redirect=1)
+| Image | YAML Label | Included Software | 
+| --------------------|---------------------|--------------------|
+| Ubuntu 22.04 | `ubuntu-latest` or `ubuntu-22.04` | [ubuntu-22.04] |
+| Ubuntu 20.04 | `ubuntu-20.04` | [ubuntu-20.04] | 
+| macOS 14 [beta] | `macos-14`, `macos-14-large`, or  `macos-14-xlarge`| [macOS-14] |
+| macOS 13 | `macos-13`, `macos-13-large`, or  `macos-13-xlarge`| [macOS-13] |
+| macOS 12 | `macos-latest`,`macos-12`, `macos-latest-large`, or `macos-12-large`| [macOS-12] | 
+| macOS 11 | `macos-11`| [macOS-11] | 
+| Windows Server 2022 | `windows-latest` or `windows-2022` | [windows-2022] |
```
Comment (muzimuzhi) [inline README.md:30]: To make linter happy.

https://github.com/actions/runner-images/actions/runs/7585086029/job/20660155612#step:4:93
```
/github/workspace/README.md:22:43 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
/github/workspace/README.md:25:51 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
/github/workspace/README.md:28:96 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
/github/workspace/README.md:29:38 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
/github/workspace/README.md:30:78 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
```

```suggestion
| Image | YAML Label | Included Software |
| --------------------|---------------------|--------------------|
| Ubuntu 22.04 | `ubuntu-latest` or `ubuntu-22.04` | [ubuntu-22.04] |
| Ubuntu 20.04 | `ubuntu-20.04` | [ubuntu-20.04] |
| macOS 14 [beta] | `macos-14`, `macos-14-large`, or  `macos-14-xlarge`| [macOS-14] |
| macOS 13 | `macos-13`, `macos-13-large`, or  `macos-13-xlarge`| [macOS-13] |
| macOS 12 | `macos-latest`,`macos-12`, `macos-latest-large`, or `macos-12-large`| [macOS-12] |
| macOS 11 | `macos-11`| [macOS-11] |
| Windows Server 2022 | `windows-latest` or `windows-2022` | [windows-2022] |
```

**Review by vlad-buhaescu-qoob — APPROVED** [review]

**Review by mikhailkoliada — APPROVED** [review]

**Review by erik-bershel — APPROVED** [review]

**Pull request merged**